internal/chart/process.go

```diff
@@ -125,12 +132,75 @@ func withVolumeMounts(vm []v1.VolumeMount) processOption {
 	}
 }
 
+// withLabels returns a function that populates Kind labels which specify a matching/unspecified deploymentVersion, a matching/unspecified
+// processName, and apiVersion v1.
+func withLabels(labels []ketchv1.MetadataItem, deploymentVersion ketchv1.DeploymentVersion) processOption {
+	return func(p *process) error {
+		for _, label := range labels {
+			if label.DeploymentVersion > 0 && int(deploymentVersion) != label.DeploymentVersion ||
```

Comment:
This is a pretty long if statement, so I think maybe adding a small comment describing what it is doing would be helpful.

Reply:
Comments added for this atrocity.

Reply:
how about 
```golang
func canBeApplied(item ketchv1.MetadataItem, processName string, version ketchv1.DeploymentVersion) bool {
	if item.Target.APIVersion != "v1" {
		return false
	}
	if item.DeploymentVersion > 0 && int(version) != item.DeploymentVersion {
		return false
	}
	if item.ProcessName != "" && processName != item.ProcessName {
		return false
	}
	return true
}
// withAnnotations returns a function that populates Kind annotations which specify a matching/unspecified deploymentVersion, a matching/unspecified
// processName, and apiVersion v1.
func withAnnotations(annotations []ketchv1.MetadataItem, deploymentVersion ketchv1.DeploymentVersion) processOption {
	return func(p *process) error {
		for _, annotation := range annotations {
			if !canBeApplied(annotation, p.Name, deploymentVersion) {
				continue
			}
                }
         }
}
```